src/lib/io/sol-iio.c

```diff
@@ -157,6 +157,8 @@ craft_filename_path(char *path, size_t size, const char *base, ...)
     len = vsnprintf(path, size, base, args);
     va_end(args);
 
+    SOL_DBG("Crafted path len: %d Buffer size: %zu", len, size);
```

Comment:
I was looking at path[size] = {}; (init all to null)

If you want to be sure you can `if (r < 0) path[0] = '\0';`